homeassistant/components/brother/sensor.py

```diff
@@ -25,356 +28,351 @@
+        value=lambda data: data.cyan_ink_remaining,
+        extra_state_attrs=lambda _: {},
     ),
     BrotherSensorEntityDescription(
-        key=ATTR_MAGENTA_INK_REMAINING,
+        key="magenta_ink_remaining",
         icon="mdi:printer-3d-nozzle",
         name="Magenta ink remaining",
         native_unit_of_measurement=PERCENTAGE,
         state_class=SensorStateClass.MEASUREMENT,
         entity_category=EntityCategory.DIAGNOSTIC,
+        value=lambda data: data.magenta_ink_remaining,
+        extra_state_attrs=lambda _: {},
     ),
     BrotherSensorEntityDescription(
-        key=ATTR_YELLOW_INK_REMAINING,
+        key="yellow_ink_remaining",
         icon="mdi:printer-3d-nozzle",
         name="Yellow ink remaining",
         native_unit_of_measurement=PERCENTAGE,
         state_class=SensorStateClass.MEASUREMENT,
         entity_category=EntityCategory.DIAGNOSTIC,
+        value=lambda data: data.yellow_ink_remaining,
+        extra_state_attrs=lambda _: {},
     ),
     BrotherSensorEntityDescription(
-        key=ATTR_UPTIME,
+        key="uptime",
         name="Uptime",
         entity_registry_enabled_default=False,
         device_class=SensorDeviceClass.TIMESTAMP,
         entity_category=EntityCategory.DIAGNOSTIC,
-        entity_class=BrotherPrinterUptimeSensor,
+        value=lambda data: data.uptime,
+        extra_state_attrs=lambda _: {},
     ),
 )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Add Brother entities from a config_entry."""
+    coordinator = hass.data[DOMAIN][DATA_CONFIG_ENTRY][entry.entry_id]
+
+    # Due to the change of the attribute name of one sensor, it is necessary to migrate
+    # the unique_id to the new one.
+    entity_registry = er.async_get(hass)
+    old_unique_id = f"{coordinator.data.serial.lower()}_b/w_counter"
+    if entity_id := entity_registry.async_get_entity_id(
+        PLATFORM, DOMAIN, old_unique_id
+    ):
+        new_unique_id = f"{coordinator.data.serial.lower()}_bw_counter"
+        _LOGGER.debug(
+            "Migrating entity %s from old unique ID '%s' to new unique ID '%s'",
+            entity_id,
+            old_unique_id,
+            new_unique_id,
+        )
+        entity_registry.async_update_entity(entity_id, new_unique_id=new_unique_id)
+
+    sensors = []
+
+    device_info = DeviceInfo(
+        configuration_url=f"http://{entry.data[CONF_HOST]}/",
+        identifiers={(DOMAIN, coordinator.data.serial)},
+        manufacturer="Brother",
+        model=coordinator.data.model,
+        name=coordinator.data.model,
+        sw_version=coordinator.data.firmware,
+    )
+
+    for description in SENSOR_TYPES:
+        if description.value(coordinator.data) is not None:
+            sensors.append(BrotherPrinterSensor(coordinator, description, device_info))
+    async_add_entities(sensors, False)
+
+
+class BrotherPrinterSensor(CoordinatorEntity, SensorEntity):
```

Comment:
Side note: We can type the coordinator in the generic coordinator entity.

```py
CoordinatorEntity[BrotherDataUpdateCoordinator]
```

Reply:
I just noticed your comment, PR is open https://github.com/home-assistant/core/pull/80197